libs/wallet/src/web3-react/hooks/useWalletMetadata.ts

```diff
@@ -125,10 +124,7 @@ export function useIsSafeApp(): boolean {
  * regardless of the connection method (WalletConnect or inside Safe as an App)
  */
 export function useIsSafeWallet(): boolean {
-  const { safeAddress } = useSafeAppsSdkInfo() || {}
-  const hasSafeInfo = !!useGnosisSafeInfo()
-
-  return !!safeAddress || hasSafeInfo
+  return !!useGnosisSafeInfo()
```

Comment:
Actually, I am checking an issue with counterfactual Safes and found out that this was added for it 

https://github.com/cowprotocol/cowswap/pull/4063/files

Does your change account for counterfactual?
Meaning, does the SDK now return the info for counterfactual Safes?

Reply:
Well, it does work now 
![image](https://github.com/cowprotocol/cowswap/assets/43217/4d34be14-236a-491b-b9e1-841c90c7eeaf)

I guess this will fix Daniel's problem, then :)

Reply:
I don't get it. It's not like this PR broke the counterfactual detection? The tests I did were with your current main release?

Reply:
This PR "fixed" counterfactual Safes (allowing them to be triggered) but at the same time "broke" the detection of them, given that they are now treated as EOAs.

Reply:
Ok, I'll play a bit more to see if I can find some middle ground.

Reply:
After debugging this with @anxolin and @alfetopito we discovered that the problem is in the useIsSmartContractWallet function and is independent of this PR:
https://github.com/cowprotocol/cowswap/blob/85a78bbfe9b0a8fe5428b25aba0b38944b10987d/libs/wallet/src/web3-react/hooks/useIsSmartContractWallet.ts#L26

With the changes in this PR though, one could call the useIsSafeWallet in the smartContractWallet and detect that this is a Safe(or a counterfactual safe). I think that the CoW team is also going to do some more detections there?

Reply:
Fix here https://github.com/cowprotocol/cowswap/pull/4553